### ISSUE TITLE
t.eval() built with slim outputs wrong predictions when input batch contains identical images

### ISSUE THREAD

**Issue by PhoenixDai** [tensorflow/tensorflow#6478]
When evaluating a model built with slim on a batch that contains identical images, the output of the batch will be wrong (mostly the outputs will be the most frequent label). For example, if my evaluation set has 421 images and my batch size is 40, I filled the last batch with 19 identical images (the 421st one) to avoid tensor shape mismatch error. Then,  the output for the 21st to the 40th image of the last batch will be the most common label in training set. If I replace the 19 images with some randomly selected images, the outputs will be correct.

### What related GitHub issues or StackOverflow threads have you found by searching the web for your problem?

None.

### Environment info
Operating System: Ubuntu 14.04.5

Installed version of CUDA and cuDNN: 
(please attach the output of `ls -l /path/to/cuda/lib/libcud*`):
![image](https://cloud.githubusercontent.com/assets/3335135/21456479/56b99382-c8f6-11e6-88ee-1b83758fb56a.png)

If installed from binary pip package, provide:

1. A link to the pip package you installed:
https://storage.googleapis.com/tensorflow/linux/gpu/tensorflow-0.11.0-cp27-none-linux_x86_64.whl

2. The output from `python -c "import tensorflow; print(tensorflow.__version__)"`.

I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcublas.so locally
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcudnn.so locally
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcufft.so locally
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcuda.so.1 locally
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcurand.so locally
0.11.0

### If possible, provide a minimal reproducible example (We usually don't have time to read hundreds of lines of your code)
The problem can be reproduced with any model built purely with slim but evaluated with output.eval(feed_dict={batch_image:test_batch}) or sess.run(output, feed_dict={batch_image:test_batch})
when the test batch contains several identical images. 

### What other attempted solutions have you tried?
Avoid identical or similar images in evaluation batch.

### Logs or other output that would be helpful
(If logs are large, please upload as attachment or provide link).

**Comment by michaelisard** [comment]
Could you follow up with a quick concrete repro case that we can try running?

**Comment by PhoenixDai** [comment]
Sure, I will wrap up a repro case in a couple of days.

**Comment by PhoenixDai** [comment]
I put together a repro case in a .ipynb file. Please place the file in the slim folder (./models/slim) so that it can find the slim dataset lib. The 4th and 7th cell set the dataset path, please point it to the converted cifar dataset tfrecord folder. Then, you should be able to run it. 

At the penultimate cell, you can see results of a batch with different images and most of the classifications match their label. However, at the last cell, when I fill the batch with identical images, the classification will always be 5. 

[slim_debug.zip](https://github.com/tensorflow/tensorflow/files/693863/slim_debug.zip)

**Comment by ppwwyyxx** [comment]
In test time you didn't seem to set batch_norm to is_training=False.
Then with an identical batch it will certainly fail.

**Comment by PhoenixDai** [comment]
Oh, thank you so much for pointing this out. I will have a try and let you know how it goes.

**Comment by PhoenixDai** [comment]
I set batch_norm to is_training=False but still have the same issue. Below is the updated evaluation/test graph:

```
graph = tf.Graph()
with graph.as_default():
  batch_images = tf.placeholder(tf.float32, (None,32,32,3))
  batch_labels = tf.placeholder(tf.float32, (None,10))    
  is_training = tf.placeholder(tf.bool)
  
  with slim.arg_scope([slim.batch_norm], is_training=is_training):
    features = classifier_features(batch_images, is_training)
    outputs, output_logits = classifier_outputs(features, dataset.num_classes)
  output_loss = tf.reduce_mean(tf.nn.softmax_cross_entropy_with_logits(output_logits, batch_labels)) 
  output_loss_reg = output_loss + tf.add_n(slim.losses.get_regularization_losses())

  train_vars = tf.trainable_variables() 
```

Am I doing it in the right way?

**Comment by sguada** [comment]
I'm not sure what your classifier_features or classifier_outputs do.
If you are using drop_out you also need to set is_training=False.

**Comment by PhoenixDai** [comment]
classifier_features and classifier_outputs are used to define the model. Please see the 3rd cell in the attached .ipynb file in my second reply to michaelisard for more details. 

Yes, I'm using drop_out and classifier_features(batch_images, is_training) will set is_training=False for drop_out

**Comment by MarkDaoust** [comment]
It looks like code in that `.ipynb` no longer runs (mostly the problems are function signatures that were updated in 1.0).  

@PhoenixDai, could you update and retry it, let us know if this is still a problem.
